crates/re_sdk/src/spawn.rs

```diff
@@ -0,0 +1,166 @@
+        self
+    }
+
+    /// Refer to field-level documentation.
+    pub fn with_executable_name(mut self, executable_name: impl AsRef<str>) -> Self {
+        self.executable_name = Some(executable_name.as_ref().to_owned());
+        self
+    }
+
+    /// Refer to field-level documentation.
+    pub fn with_executable_path(mut self, executable_path: impl AsRef<str>) -> Self {
+        self.executable_path = Some(executable_path.as_ref().to_owned());
+        self
+    }
+}
+
+impl SpawnOptions {
+    /// Resolves the final port value.
+    pub fn port(&self) -> u16 {
+        self.port.unwrap_or(9876)
+    }
+
+    /// Resolves the final connect address value.
+    pub fn connect_addr(&self) -> std::net::SocketAddr {
+        std::net::SocketAddr::new("127.0.0.1".parse().unwrap(), self.port())
+    }
+
+    /// Resolves the final listen address value.
+    pub fn listen_addr(&self) -> std::net::SocketAddr {
+        std::net::SocketAddr::new("0.0.0.0".parse().unwrap(), self.port())
+    }
+
+    /// Resolves the final memory limit value.
+    pub fn memory_limit(&self) -> String {
+        self.memory_limit.as_deref().unwrap_or("75%").to_owned()
+    }
+
+    /// Resolves the final executable path.
+    pub fn executable_path(&self) -> String {
+        // NOTE: No need for .exe extension on windows.
+        const RERUN_BINARY: &str = "rerun";
+
+        if let Some(path) = self.executable_path.as_deref() {
+            return path.to_owned();
+        }
+
+        self.executable_name
+            .as_deref()
+            .unwrap_or(RERUN_BINARY)
+            .to_owned()
+    }
+}
+
+/// Spawns a new Rerun Viewer process ready to listen for TCP connections.
+///
+/// Refer to [`SpawnOptions`]'s documentation for configuration options.
+///
+/// This only starts a Viewer process: if you'd like to connect to it and start sending data, refer
+/// to [`crate::RecordingStream::connect`] or use [`crate::RecordingStream::spawn`] directly.
+pub fn spawn(opts: &SpawnOptions) -> std::io::Result<()> {
+    use std::{net::TcpStream, process::Command, time::Duration};
+
+    // NOTE: It's indented on purpose, it just looks better and reads easier.
+    const EXECUTABLE_NOT_FOUND: &str = //
+    "
+    Couldn't find the Rerun Viewer executable in your PATH.
+
+    You can install binary releases of the Rerun Viewer using any of the following methods:
+    * Binary download with `cargo`: `cargo binstall rerun-cli` (see https://github.com/cargo-bins/cargo-binstall)
+    * Build from source with `cargo`: `cargo install rerun-cli` (requires Rust 1.72+)
+    * Direct download from our release assets: https://github.com/rerun-io/rerun/releases/latest/
+    * Or together with the Rerun Python SDK:
+      * Pip: `pip3 install rerun-sdk`
+      * Conda: `conda install -c conda-forge rerun-sdk`
+      * Binary download with `pixi`: `pixi global install rerun-sdk` (see https://prefix.dev/docs/pixi/overview)
```

Comment:
It's probably better to link to `docs/content/getting-started/installing-viewer.md` instead of repeating ourselves

Reply:
I'm conflicted about this... I don't like duplication but on the other hand I'm not sure I want users to have to leave their terminal and read some external docs in the middle of their workflow.

Reply:
I could `include!(installing-viewer.md)` but it's really not optimized for terminal printing.

Reply:
Let's do both then - repeat the "best" install options, and then link to the guide for more details ?

Reply:
Actually I need the current context to know the correct viewer version to install (https://github.com/rerun-io/rerun/issues/4001).

Reply:
Yeah both sounds good